plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,69 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+	private const string MAGIC_VALUE = "BOOT";
+
+	private string _disc_id;
+	public string disc_id {
+		get { return _disc_id; }
+	}
+
+	private File file;
+	private StringInputStream stream;
+
+  public PlayStationHeader (File file) {
+		this.file = file;
+		stream = new StringInputStream (file);
+  }
+
+	public void check_validity () throws PlayStationError {
+		var offset = get_header_offset ();
+
+		string data;
+		try {
```

Comment:
It's probably not needed to encapsulate this in try catch blocks in the end, just let the error flow.